admin/class-boldgrid-backup-admin-task.php

```diff
@@ -0,0 +1,292 @@
+	 */
+	private $helper;
+
+	/**
+	 * The task id.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var string
+	 */
+	private $id;
+
+	/**
+	 * The time this task was started.
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var int
+	 */
+	private $started_at;
+
+	/**
+	 * The task type.
+	 *
+	 * For example: backup
+	 *
+	 * @since SINCEVERSION
+	 * @access private
+	 * @var string
+	 */
+	private $type;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function __construct() {
+		$this->helper = new Boldgrid_Backup_Admin_Task_Helper();
+	}
+
+	/**
+	 * Mark this task as being complete.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function end() {
+		$this->completed_at = time();
+
+		$this->update();
+	}
+
+	/**
+	 * Get the properties of this class.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return array
+	 */
+	public function get() {
+		return [
+			'id'           => $this->id,
+			'type'         => $this->type,
+			'created_at'   => empty( $this->date_format ) ? $this->created_at : date( $this->date_format, $this->created_at ),
+			'started_at'   => empty( $this->started_at ) ?
+				null :
+				( empty( $this->date_format ) ? $this->started_at : date( $this->date_format, $this->started_at ) ),
+			'completed_at' => empty( $this->completed_at ) ?
+				null :
+				( empty( $this->date_format ) ? $this->completed_at : date( $this->date_format, $this->completed_at ) ),
+			'status'       => $this->get_status(),
+			'data'         => $this->data,
+		];
+	}
+
+	/**
+	 * Get a specific value from the data.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param  string $key     The key to get.
+	 * @param  mixed  $default The default value to return.
+	 * @return mixed
+	 */
+	public function get_data( $key, $default = false ) {
+		return isset( $this->data[ $key ] ) ? $this->data[ $key ] : $default;
+	}
+
+	/**
+	 * Get our task id.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return string
+	 */
+	public function get_id() {
+		return $this->id;
+	}
+
+	/**
+	 * Get the status of this task.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return string
+	 */
+	public function get_status() {
+		if ( empty( $this->started_at ) ) {
+			$status = 'pending';
+		} elseif ( empty( $this->completed_at ) ) {
+			$status = 'in_progress';
+		} else {
+			$status = 'done';
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Initialize a new task.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param  array $args An array of args.
+	 * @return bool        True if this task was initialized.
+	 */
+	public function init( $args ) {
+		$this->id           = ! empty( $args['id'] ) ? $args['id'] : $this->set_id();
+		$this->created_at   = ! empty( $args['created_at'] ) ? $args['created_at'] : time();
+		$this->started_at   = ! empty( $args['started_at'] ) ? $args['started_at'] : null;
+		$this->completed_at = ! empty( $args['completed_at'] ) ? $args['completed_at'] : null;
+		$this->data         = ! empty( $args['data'] ) ? $args['data'] : [];
```

Comment:
Arrays should be declared with long array syntax per WordPress coding standards
https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#declaring-arrays

Reply:
See similar comment above.